src/main/java/org/stellar/sdk/responses/PageDeserializer.java

```diff
@@ -45,6 +46,7 @@ public Page<E> deserialize(JsonElement json, Type typeOfT, JsonDeserializationCo
             .registerTypeAdapter(EffectResponse.class, new EffectDeserializer())
             .registerTypeAdapter(LiquidityPoolResponse.class, new LiquidityPoolDeserializer())
             .registerTypeAdapter(TransactionResponse.class, new TransactionDeserializer())
+            .registerTypeAdapter(LiquidityPoolID.class, new LiquidityPoolIDDeserializer())
```

Comment:
Interesting, I see the same type adapter registration for LiquidityPoolID done up at the main gson instance used by all response handlers:

https://github.com/vinamogit/java-stellar-sdk/blob/fix-lp-id-deserialization-in-page/src/main/java/org/stellar/sdk/responses/GsonSingleton.java#L39

Do you know why this is not getting used by gson to resolve the LiquidityPoolID in `TradeResponse` during unmarshaling? It looks TradesRequestBuilder.execute uses the ResponseHandler which in turn uses that gson instance which includes the type registration. If re-registering here at page deser is needed, so be it, nice find!

Reply:
I don't know anything about Gson, my guess is that it's either because of generic type or because a new Gsonbuilder is used.

Reply:
The thing is that there are several Gson instances, each need to know the deserializers.
We could avoid the double maintenance between GsonSingleton and PageDeserializer by using the JsonAdapter annotation on all the types which are not Page<T>.

The only type that would need to be registered with both GsonBuilders will be the TransactionDeserializer since it uses a third instance of Gson to reconciliate the memo with its memo_type.

I published a branch for comparison:
https://github.com/vinamogit/java-stellar-sdk/commit/fe9c9de32aaedc964cc60bc42718f8f15bcc8267